SYCL/Scheduler/AllocaCleanup.cpp

```diff
@@ -0,0 +1,97 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_PI_TRACE=2 %GPU_RUN_PLACEHOLDER %t.out 2>&1 %GPU_CHECK_PLACEHOLDER
+
+#include <sys/resource.h>
+#include <oneapi/dpl/execution>
+#include <oneapi/dpl/numeric>
+#include <iostream>
+#include <vector>
+#include "CL/sycl.hpp"
+
+template<typename RealType>
+void cross_product (sycl::queue& q, 
+                    int n, int p,
+                    sycl::buffer<RealType, 1> data_buf,
+                    sycl::buffer<RealType, 1> w_buf,
+                    sycl::buffer<RealType, 1> mean_buf, 
+                    sycl::buffer<RealType, 1> cross_product_buf)
+{
+    std::cout << '.';
+    sycl::buffer<RealType, 1> helper_buf(n * p, sycl::no_init); // props
+
+    q.submit([&] (auto &h) {
+        sycl::accessor data_acc(data_buf, h, sycl::read_only);
+        sycl::accessor helper_acc(helper_buf, h, sycl::write_only);
+        sycl::accessor mean_acc(mean_buf, h, sycl::read_only);
+        h.parallel_for(sycl::range<1>(n * p), [=] (sycl::item<1> item) {
+            int ij = item.get_id();
+            helper_acc[ij] = data_acc[ij] - mean_acc[ij / n];
+        });
+    });
+    q.wait();
+
+    std::vector<sycl::buffer<RealType, 1>> sub_helper; // props
+    for (int i = 0; i < p; i++) {
+        sub_helper.push_back(sycl::buffer<RealType, 1>(helper_buf, i*n, n)); 
+    }
+
+    auto policy = oneapi::dpl::execution::make_device_policy(q);
+    auto w_begin = oneapi::dpl::begin(w_buf);
```

Comment:
I believe it's better to avoid dependencies on non SYCL APIs in these tests.
Tagging @vladimirlaz

Reply:
It is better to avoid extra dependencies in these tests? Is it possible to remove the use of DPL?

Reply:
Added is the reproducer for the sporadic segfaults fixed in https://github.com/intel/llvm/pull/4573. I agree that having dependencies to other moving components in SYCL tests is not a good idea, however this was the only reproducer I have at hand. I'll check to see if the issue reproduces even after removing the DPL dependencies.